conda-store-server/tests/test_server.py

```diff
@@ -396,6 +397,77 @@ def test_create_specification_auth_env_name_too_long(testclient, celery_worker,
         assert False, f"failed to update status"
 
 
+@pytest.fixture
+def win_extended_length_prefix(request):
+    # Overrides the attribute before other fixtures are called
+    import conda_store_server
+    conda_store_server.app.CondaStore.win_extended_length_prefix = request.param
+    yield request.param
+
+
+@pytest.mark.skipif(sys.platform != "win32", reason="tests a Windows issue")
+@pytest.mark.parametrize('win_extended_length_prefix', [True, False], indirect=True)
+@pytest.mark.extended_prefix
+def test_create_specification_auth_extended_prefix(win_extended_length_prefix, testclient, celery_worker, authenticate):
+    # Adds padding to cause an error if the extended prefix is not enabled
+    namespace = "default" + 'A' * 10
+    environment_name = "pytest"
+
+    # The debugpy 1.8.0 package was deliberately chosen because it has long
+    # paths internally, which causes issues on Windows due to the path length
+    # limit
+    response = testclient.post(
+        "api/v1/specification",
+        json={
+            "namespace": namespace,
+            "specification": json.dumps({
+                "name": environment_name,
+                "channels": ["conda-forge"],
+                "dependencies": ["debugpy==1.8.0"],
+                "variables": None,
+                "prefix": None,
+                "description": "test"
+            }),
+        },
+        timeout=30,
+    )
+    response.raise_for_status()
+
+    r = schema.APIPostSpecification.parse_obj(response.json())
+    assert r.status == schema.APIStatus.OK
+    build_id = r.data.build_id
+
+    # Try checking that the status is 'FAILED'
+    is_updated = False
+    for _ in range(30):
+        time.sleep(5)
+
+        # check for the given build
+        response = testclient.get(f"api/v1/build/{build_id}", timeout=30)
+        response.raise_for_status()
+
+        r = schema.APIGetBuild.parse_obj(response.json())
+        assert r.status == schema.APIStatus.OK
+        assert r.data.specification.name == environment_name
+        if r.data.status in ["QUEUED", "BUILDING"]:
+            continue  # checked too fast, try again
+
+        if win_extended_length_prefix:
+            assert r.data.status == "COMPLETED"
+        else:
+            assert r.data.status == "FAILED"
+            response = testclient.get(f"api/v1/build/{build_id}/logs", timeout=30)
+            response.raise_for_status()
+            assert "[WinError 206] The filename or extension is too long" in response.text
+
+        is_updated = True
+        break
+
+    # If we're here, the task didn't update the status on failure
+    if not is_updated:
+        assert False, f"failed to get status"
+
+
```

Comment:
Note for reviewers: the test above is based on `test_create_specification_auth_env_name_too_long`. I deliberately didn't move common bits into a separate function because I think the current version is more readable